diff -uNrp a/cmake/modules/FindFFMPEG.cmake b/cmake/modules/FindFFMPEG.cmake
--- a/cmake/modules/FindFFMPEG.cmake	2026-06-21 08:53:12.000000000 +0100
+++ b/cmake/modules/FindFFMPEG.cmake	2026-08-22 13:22:20.912288881 +0100
@@ -281,13 +281,13 @@ else()
   # have latest version to properly track rebuiling.
   if(KODI_DEPENDSBUILD OR (WIN32 OR WINDOWS_STORE))
     # required ffmpeg library versions - tools/depends/target/ffmpeg versions
-    set(REQUIRED_FFMPEG_VERSION 8.1.2)
-    set(_avutil_ver "=60.26.102")
-    set(_avcodec_ver "=62.28.102")
-    set(_avformat_ver "=62.12.102")
-    set(_avfilter_ver "=11.14.102")
-    set(_swscale_ver "=9.5.102")
-    set(_swresample_ver "=6.3.102")
+    set(REQUIRED_FFMPEG_VERSION 9.0.1)
+    set(_avutil_ver "=61.1.101")
+    set(_avcodec_ver "=63.1.101")
+    set(_avformat_ver "=63.1.101")
+    set(_avfilter_ver "=12.1.101")
+    set(_swscale_ver "=10.1.101")
+    set(_swresample_ver "=7.1.101")
     set(_postproc_ver "=59.1.100")
   else()
     # required ffmpeg library versions - minimum supported API compat versions
diff -uNrp a/tools/depends/target/ffmpeg/FFMPEG-VERSION b/tools/depends/target/ffmpeg/FFMPEG-VERSION
--- a/tools/depends/target/ffmpeg/FFMPEG-VERSION	2026-06-21 08:53:12.000000000 +0100
+++ b/tools/depends/target/ffmpeg/FFMPEG-VERSION	2026-08-22 13:22:20.912392799 +0100
@@ -1,4 +1,4 @@
 LIBNAME=ffmpeg
-VERSION=8.1.2
+VERSION=9.0.1
 ARCHIVE=$(LIBNAME)-$(VERSION).tar.xz
-SHA512=b3adc16fe426217bb607da01c4137cee0a9788fc08e077874336db185d2b7287746a7dc94cf0181ea92cd8afcdb06094ee9456e2986112354f84538cb9a5ed0b
+SHA512=be8cdfca64886f2ec5cd2359f8db6a998ee759d105b10ade9b2a7b640356ad56ab0ff50218fce7b6312462f15f98d90ac9d197a3b02a23771754d2714fec628d
