diff -uNrp a/xbmc/interfaces/python/swig.cpp b/xbmc/interfaces/python/swig.cpp
--- a/xbmc/interfaces/python/swig.cpp	2024-04-06 03:29:22.000000000 +0100
+++ b/xbmc/interfaces/python/swig.cpp	2024-06-13 23:06:04.977533658 +0100
@@ -75,6 +75,9 @@ namespace PythonBindings
 #if PY_VERSION_HEX >= 0x030C00A1
       0,
 #endif
+#if PY_VERSION_HEX >= 0x030D00B2
+      0,
+#endif
     };
 
     static int size = (long*)&(py_type_object_header.tp_name) - (long*)&py_type_object_header;
