--- a/xbmc/cores/DllLoader/exports/wrapper.c	2018-03-04 03:21:28.000000000 -0600
+++ b/xbmc/cores/DllLoader/exports/wrapper.c	2018-03-15 10:04:56.995231658 -0500
@@ -45,9 +45,6 @@
 typedef off64_t   __off64_t;
 typedef fpos_t    fpos64_t;
 #define stat64    stat
-#if defined(TARGET_DARWIN) || defined(TARGET_ANDROID)
-#define _G_va_list va_list
-#endif
 #endif
 
 #ifdef TARGET_POSIX
@@ -485,7 +482,7 @@
   return res;
 }
 
-int __wrap___vfprintf_chk(FILE* stream, int flag, const char *format, _G_va_list ap)
+int __wrap___vfprintf_chk(FILE* stream, int flag, const char *format, va_list ap)
 {
   return dll_vfprintf(stream, format, ap);
 }
