--- a/xbmc/windowing/wayland/Connection.h	2023-01-15 03:13:34.000000000 -0600
+++ b/xbmc/windowing/wayland/Connection.h	2023-01-17 07:30:23.626806369 -0600
@@ -8,6 +8,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <memory>
 
 #include <wayland-client.hpp>
--- a/cmake/scripts/linux/ExtraTargets.cmake	2023-01-15 03:13:34.000000000 -0600
+++ b/cmake/scripts/linux/ExtraTargets.cmake	2023-01-17 07:52:34.171899607 -0600
@@ -24,6 +24,7 @@
                     "${WAYLAND_PROTOCOLS_DIR}/unstable/idle-inhibit/idle-inhibit-unstable-v1.xml")
   add_custom_command(OUTPUT "${WAYLAND_EXTRA_PROTOCOL_GENERATED_DIR}/wayland-extra-protocols.hpp" "${WAYLAND_EXTRA_PROTOCOL_GENERATED_DIR}/wayland-extra-protocols.cpp"
                      COMMAND "${WAYLANDPP_SCANNER}" ${PROTOCOL_XMLS} "${WAYLAND_EXTRA_PROTOCOL_GENERATED_DIR}/wayland-extra-protocols.hpp" "${WAYLAND_EXTRA_PROTOCOL_GENERATED_DIR}/wayland-extra-protocols.cpp"
+                     COMMAND "sed" "-i" "'4i#include <cstdint>'" "${WAYLAND_EXTRA_PROTOCOL_GENERATED_DIR}/wayland-extra-protocols.hpp"
                      DEPENDS "${WAYLANDPP_SCANNER}" ${PROTOCOL_XMLS}
                      COMMENT "Generating wayland-protocols C++ wrappers")
 
--- a/xbmc/cores/RetroPlayer/cheevos/Cheevos.h	2023-01-15 03:13:34.000000000 -0600
+++ b/xbmc/cores/RetroPlayer/cheevos/Cheevos.h	2023-01-17 07:58:48.447329727 -0600
@@ -10,6 +10,7 @@
 
 #include "RConsoleIDs.h"
 
+#include <cstdint>
 #include <map>
 #include <string>
 
--- a/xbmc/pictures/Picture.h	2023-01-15 03:13:34.000000000 -0600
+++ b/xbmc/pictures/Picture.h	2023-01-17 08:12:50.692935868 -0600
@@ -11,6 +11,7 @@
 #include "pictures/PictureScalingAlgorithm.h"
 #include "utils/Job.h"
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
--- a/xbmc/utils/CSSUtils.cpp	2023-01-15 03:13:34.000000000 -0600
+++ b/xbmc/utils/CSSUtils.cpp	2023-01-17 08:18:51.028020811 -0600
@@ -8,6 +8,7 @@
 
 #include "CSSUtils.h"
 
+#include <cstdint>
 #include <string>
 
 namespace
--- a/xbmc/utils/CharArrayParser.h	2023-01-15 03:13:34.000000000 -0600
+++ b/xbmc/utils/CharArrayParser.h	2023-01-17 08:33:26.240155696 -0600
@@ -8,6 +8,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string>
 
 /*!
--- a/xbmc/utils/StreamUtils.h	2023-01-15 03:13:34.000000000 -0600
+++ b/xbmc/utils/StreamUtils.h	2023-01-17 08:39:29.936429541 -0600
@@ -8,6 +8,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string>
 
 static constexpr int MP4_BOX_HEADER_SIZE = 8;
--- a/xbmc/utils/DRMHelpers.h	2023-01-15 03:13:34.000000000 -0600
+++ b/xbmc/utils/DRMHelpers.h	2023-01-17 08:48:09.207127805 -0600
@@ -8,6 +8,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string>
 
 namespace DRMHELPERS
--- a/xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.h	2023-01-15 03:13:34.000000000 -0600
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/ColorManager.h	2023-01-17 08:53:27.315723763 -0600
@@ -12,6 +12,7 @@
 #include <lcms2.h>
 #endif
 
+#include <cstdint>
 #include <string>
 
 extern "C"
--- a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPIGL.h	2023-01-15 03:13:34.000000000 -0600
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPIGL.h	2023-01-17 08:55:33.010153456 -0600
@@ -11,6 +11,7 @@
 #include "VaapiEGL.h"
 #include "cores/VideoPlayer/VideoRenderers/LinuxRendererGL.h"
 
+#include <cstdint>
 #include <memory>
 
 namespace VAAPI
--- a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.h	2023-01-15 03:13:34.000000000 -0600
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.h	2023-01-17 08:58:59.852480329 -0600
@@ -9,6 +9,7 @@
 #pragma once
 
 #include <array>
+#include <cstdint>
 
 #if defined(HAS_GL)
 // always define GL_GLEXT_PROTOTYPES before include gl headers
--- a/xbmc/platform/Filesystem.h	2023-01-15 03:13:34.000000000 -0600
+++ b/xbmc/platform/Filesystem.h	2023-01-17 09:00:49.412722028 -0600
@@ -8,6 +8,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string>
 #include <system_error>
 namespace KODI
