--- a/xbmc/network/WebServer.cpp	2020-07-05 15:50:53.000000000 -0500
+++ b/xbmc/network/WebServer.cpp	2020-07-06 17:24:42.764074424 -0500
@@ -510,7 +510,7 @@
     return;
 
   // otherwise we can use MHD's POST processor
-  connectionHandler->postprocessor = MHD_create_post_processor(request.connection, MAX_POST_BUFFER_SIZE, &CWebServer::HandlePostField, static_cast<void*>(connectionHandler));
+  connectionHandler->postprocessor = MHD_create_post_processor(request.connection, MAX_POST_BUFFER_SIZE, (MHD_PostDataIterator)(&CWebServer::HandlePostField), static_cast<void*>(connectionHandler));
 
   // MHD doesn't seem to be able to handle this post request
   if (connectionHandler->postprocessor == nullptr)
@@ -1155,7 +1155,7 @@
                           port,
                           0,
                           0,
-                          &CWebServer::AnswerToConnection,
+                          (MHD_AccessHandlerCallback)(&CWebServer::AnswerToConnection),
                           this,
 
                           MHD_OPTION_CONNECTION_LIMIT, 512,
@@ -1182,7 +1182,7 @@
                           port,
                           0,
                           0,
-                          &CWebServer::AnswerToConnection,
+                          (MHD_AccessHandlerCallback)(&CWebServer::AnswerToConnection),
                           this,
 
                           MHD_OPTION_CONNECTION_LIMIT, 512,
