From a80cbc027580489cb3fbaca6fa5be9261d61131a Mon Sep 17 00:00:00 2001
From: David Heim <david@thymoze.com>
Date: Tue, 14 Dec 2021 18:28:49 +0100
Subject: [PATCH] ffmpeg: set channel count on avframe

---
 xbmc/cores/AudioEngine/Encoders/AEEncoderFFmpeg.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/cores/AudioEngine/Encoders/AEEncoderFFmpeg.cpp b/xbmc/cores/AudioEngine/Encoders/AEEncoderFFmpeg.cpp
index 143ced5595a2d..f54d27abafbdc 100644
--- a/xbmc/cores/AudioEngine/Encoders/AEEncoderFFmpeg.cpp
+++ b/xbmc/cores/AudioEngine/Encoders/AEEncoderFFmpeg.cpp
@@ -259,6 +259,7 @@ int CAEEncoderFFmpeg::Encode(uint8_t *in, int in_size, uint8_t *out, int out_siz
   frame->nb_samples = m_CodecCtx->frame_size;
   frame->format = m_CodecCtx->sample_fmt;
   frame->channel_layout = m_CodecCtx->channel_layout;
+  frame->channels = m_CodecCtx->channels;
 
   avcodec_fill_audio_frame(frame, m_CodecCtx->channels, m_CodecCtx->sample_fmt,
                     in, in_size, 0);
