--- a/xbmc/profiles/dialogs/GUIDialogProfileSettings.cpp	2019-11-16 05:12:49.000000000 -0600
+++ b/xbmc/profiles/dialogs/GUIDialogProfileSettings.cpp	2019-12-22 13:15:26.037386276 -0600
@@ -8,6 +8,7 @@
 
 #include "GUIDialogProfileSettings.h"
 
+#include <cassert>
 #include <utility>
 
 #include "dialogs/GUIDialogFileBrowser.h"
--- a/xbmc/windowing/GraphicContext.cpp	2019-11-16 05:12:49.000000000 -0600
+++ b/xbmc/windowing/GraphicContext.cpp	2019-12-22 13:34:42.062192068 -0600
@@ -6,6 +6,8 @@
  *  See LICENSES/README.md for more information.
  */
 
+#include <cassert>
+
 #include "GraphicContext.h"
 #include "WinSystem.h"
 #include "Application.h"
