--- kodi-18.0b4/cmake/scripts/common/Macros.cmake.orig	2018-10-23 04:00:54.000000000 -0500
+++ kodi-18.0b4/cmake/scripts/common/Macros.cmake	2018-11-03 13:50:45.626391925 -0500
@@ -331,6 +331,9 @@
 
 # helper macro to set modified variables in parent scope
 macro(export_dep)
+  #message(WARNING "Original libraries to link: ${DEPLIBS}")
+  string(REGEX REPLACE "tspi" "" DEPLIBS "${DEPLIBS}")
+  #message(WARNING "Adding these libraries to link: ${DEPLIBS}")
   set(SYSTEM_INCLUDES ${SYSTEM_INCLUDES} PARENT_SCOPE)
   set(DEPLIBS ${DEPLIBS} PARENT_SCOPE)
   set(DEP_DEFINES ${DEP_DEFINES} PARENT_SCOPE)
