--- a/xbmc/Application.cpp	2020-07-05 15:50:53.000000000 -0500
+++ b/xbmc/Application.cpp	2020-07-06 17:37:10.892852743 -0500
@@ -439,7 +439,7 @@ bool CApplication::Create(const CAppPara
   init_emu_environ();
 
   CLog::Log(LOGINFO, "-----------------------------------------------------------------------");
-  CLog::Log(LOGINFO, "Starting %s (%s). Platform: %s %s %d-bit", CSysInfo::GetAppName().c_str(),
+  CLog::Log(LOGINFO, "Starting %s from RPM Fusion (%s). Platform: %s %s %d-bit", CSysInfo::GetAppName().c_str(),
             CSysInfo::GetVersion().c_str(), g_sysinfo.GetBuildTargetPlatformName().c_str(),
             g_sysinfo.GetBuildTargetCpuFamily().c_str(), g_sysinfo.GetXbmcBitness());
 
