--- a/xbmc/screensavers/rsxs-0.9/lib/stdbool_.h	2015-06-17 00:49:22.686927610 -0500
+++ b/xbmc/screensavers/rsxs-0.9/lib/stdbool_.h	2015-06-17 00:51:40.145624074 -0500
@@ -78,7 +78,7 @@
 typedef bool _Bool;
 # endif
 #else
-# if !defined __GNUC__
+# if !defined __GNUC__ || __GNUC__ >= 5
    /* If @HAVE__BOOL@:
         Some HP-UX cc and AIX IBM C compiler versions have compiler bugs when
         the built-in _Bool type is used.  See
@@ -96,6 +96,8 @@
           "warning(1185): enumerated type mixed with another type".
         The only benefit of the enum type, debuggability, is not important
         with these compilers.  So use 'signed char' and no typedef.  */
+   /* The default C mode changed from C89 to C11, which removes the _Bool
+      type.  */
 #  define _Bool signed char
 enum { false = 0, true = 1 };
 # else
