--- a/Makefile.in	2016-06-30 14:18:13.000000000 -0500
+++ b/Makefile.in	2016-07-30 12:23:17.468082870 -0500
@@ -10,8 +10,7 @@
 	tools/EventClients
 
 DVDPCODECS_DIRS= \
-	lib \
-	lib/libdvd
+	lib
 
 VideoPlayer_ARCHIVES=xbmc/cores/VideoPlayer/VideoPlayer.a \
                    xbmc/cores/VideoPlayer/DVDCodecs/DVDCodecs.a \
@@ -343,7 +342,6 @@
 	$(MAKE) -C lib/addons/library.xbmc.pvr
 	$(MAKE) -C lib/addons/library.kodi.inputstream
 dvdpcodecs: dllloader
-	$(MAKE) -C lib/libdvd
 
 dvdpextcodecs:
 

